client/extensions/woocommerce/app/order/index.js

```diff
@@ -42,43 +45,90 @@ class Order extends Component {
 		if ( newProps.orderId !== this.props.orderId || newProps.siteId !== this.props.siteId ) {
 			this.props.fetchOrder( newProps.siteId, newProps.orderId );
 			this.props.fetchNotes( newProps.siteId, newProps.orderId );
-		} else if ( newProps.order && this.props.order && newProps.order.status !== this.props.order.status ) {
+		} else if (
+			newProps.order &&
+			this.props.order &&
+			newProps.order.status !== this.props.order.status
+		) {
 			// A status change should force a notes refresh
 			this.props.fetchNotes( newProps.siteId, newProps.orderId, true );
 		}
 	}
 
-	onUpdate = ( order ) => {
-		// Merge the new order updates into the existing order updates
-		this.setState( ( prevState ) => {
-			const updatedOrder = { ...prevState.order, ...order };
-			return { order: updatedOrder };
-		} );
-	}
+	// Put this order into the editing state
+	toggleEditing = () => {
+		const { siteId, orderId } = this.props;
+		if ( siteId ) {
+			this.props.editOrder( siteId, { id: orderId } );
+		}
+	};
 
+	// Clear this order's edits, takes it out of edit state
```

Comment:
Since the orderEdit state is only tied to a `siteId` and not an `orderId` - is there any chance we could get a polluted `orderEdit` state if someone edits an order here, then closes out the tab, or never executes a save or cancel action?